src/jdk.jlink/share/classes/jdk/tools/jlink/builder/HermeticImageBuilder.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2025 Red Hat, Inc.
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.tools.jlink.builder;
+
+import java.io.BufferedOutputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import jdk.tools.jlink.internal.ExecutableImage;
+import jdk.tools.jlink.internal.Platform;
+import jdk.tools.jlink.plugin.ResourcePool;
+
+public final class HermeticImageBuilder implements ImageBuilder {
+
+    private final Platform platform;
+    private final Path hermeticImage;
+    private final Set<String> modules = new HashSet<String>();
+
+    public HermeticImageBuilder(Platform platform, Path hermeticImage) {
```

Comment:
How about making `HermeticImageBuilder(Platform platform, Path hermeticImage)` argument ordering to be more consistent with `DefaultExecutableImage(Path home, Set<String> modules, Platform p)`:

```
HermeticImageBuilder(Path hermeticImage, Platform platform)
```